### PR TITLE
Config max request header size

### PR DESCRIPTION
**Pull request by guscarreon**
Implements #168 by adding a config field to specify the http request max header size in bytes.

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -201,6 +203,12 @@ func (cfg *RequestLimits) validateAndLog() {
 	} else {
 		log.Fatalf("invalid config.request_limits.max_num_values: %d. Value cannot be negative.", cfg.MaxNumValues)
 	}
+
+	if cfg.MaxHeaderSize >= 0 {
+		log.Infof("config.request_limits.max_header_size_bytes: %d", cfg.MaxHeaderSize)
+	} else {
+		log.Fatalf("invalid config.request_limits.max_header_size_bytes: %d. Value cannot be negative.", cfg.MaxHeaderSize)
```
Comment (AlexBVolcy) [inline config/config.go:210]: Can you add a quick test where the value is negative?

File: server/server.go [hunk 2]
```diff
@@ -74,20 +74,36 @@ func Listen(cfg config.Configuration, publicHandler http.Handler, adminHandler h
 	return
 }
 
+// newAdminServer returns an http.Server with the configured with the AdminPort and
```
Comment (AlexBVolcy) [inline server/server.go:77]: Nitpick: I think you can get rid of the extra "with the" --> `newAdminServer returns an http.Server with the configured AdminPort and...`

**Comment by bsardo** [comment]
Looks good. I'm just waiting to see @AlexBVolcy's comments addressed.

File: config/config.go [hunk 3]
```diff
@@ -95,6 +95,7 @@ func setConfigDefaults(v *viper.Viper) {
 	v.SetDefault("request_limits.max_size_bytes", utils.REQUEST_MAX_SIZE_BYTES)
 	v.SetDefault("request_limits.max_num_values", utils.REQUEST_MAX_NUM_VALUES)
 	v.SetDefault("request_limits.max_ttl_seconds", utils.REQUEST_MAX_TTL_SECONDS)
+	v.SetDefault("request_limits.max_header_size_bytes", 0)
```
Comment (SyntaxNode) [inline config/config.go:98]: Does 0 = no limit?

Comment (guscarreon) [reply config/config.go:98]: If zero, the `http.Server` is created with a the `http` library's `DefaultMaxHeaderBytes` constant that equals 1 MB

**Review by AlexBVolcy — APPROVED** [review]

**Review by AlexBVolcy — APPROVED** [review]

**Review by bsardo — APPROVED** [review]
